editor/editor_dock_manager.cpp

```diff
@@ -52,6 +53,115 @@ enum class TabStyle {
 
 EditorDockManager *EditorDockManager::singleton = nullptr;
 
+// Implemented in input(..) as nearly every single child consumes the mouse input.
+void EditorDockTabContainer::input(const Ref<InputEvent> &p_event) {
+	ERR_FAIL_COND(p_event.is_null());
+
+	Ref<InputEventMouse> me = p_event;
+	if (me.is_valid() && is_dragging_dock) {
+		if (mouse_inside != get_global_rect().has_point(me->get_position())) {
+			mouse_inside = !mouse_inside;
+			queue_redraw();
+		}
+	}
+}
+
+void EditorDockTabContainer::_notification(int p_what) {
+	switch (p_what) {
+		case EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED: {
+			int _corner_radius = EDSCALE * EDITOR_GET("interface/theme/corner_radius").operator int();
+			if (_corner_radius != dock_drop_highlight->get_corner_radius(CORNER_TOP_LEFT)) {
+				dock_drop_highlight->set_corner_radius_all(_corner_radius);
+				if (mouse_inside) {
+					queue_redraw();
+				}
+			}
+		} break;
+
+		case NOTIFICATION_POSTINITIALIZE:
+		case NOTIFICATION_THEME_CHANGED: {
+			valid_drop_color = get_theme_color(SNAME("accent_color"), EditorStringName(Editor));
+			invalid_drop_color = get_theme_color(SNAME("error_color"), EditorStringName(Editor));
+			int _dock_margin = get_theme_constant(SNAME("base_margin"), EditorStringName(Editor));
+
+			if (dock_margin != _dock_margin) {
+				dock_margin = _dock_margin;
```

Comment:
Use a more explicit variable name:

```suggestion
			int new_dock_margin = get_theme_constant(SNAME("base_margin"), EditorStringName(Editor));

			if (dock_margin != new_dock_margin) {
				dock_margin = new_dock_margin;
```